src/asm3/medical.py

```diff
@@ -1252,28 +1307,41 @@ def insert_profile_from_form(dbo: Database, username: str, post: PostedData) ->
     timingrule = post.integer("timingrule")
     timingrulenofrequencies = post.integer("timingrulenofrequencies")
     timingrulefrequency = post.integer("timingrulefrequency")
-    totalnumberoftreatments = post.integer("totalnumberoftreatments")
-    treatmentrule = post.integer("treatmentrule")
-    singlemulti = post.integer("singlemulti")
-    if singlemulti == 0:
-        timingrule = 0
-        timingrulenofrequencies = 0
-        timingrulefrequency = 0
-        totalnumberoftreatments = 1
-    if totalnumberoftreatments == 0:
-        totalnumberoftreatments = 1
-    if treatmentrule != 0:
-        totalnumberoftreatments = 0
+    customtimingrule = post["customtiming"].strip()
+
+    # Remove trailing comma if it exists
+    if len(customtimingrule) > 0:
+        while customtimingrule[-1] == ",":
+            customtimingrule = customtimingrule[:-1]
+    
+    if post.integer("singlemulti") == 2:
+        totalnumberoftreatments = len(customtimingrule.split(","))
```

Comment:
I'd make some constants at the top and use those here to make this code a bit more readable: 

Eg: TREATMENT_SINGLE = 0, TREATMENT_MULTI = 1, TREATMENT_CUSTOM = 2 

```if post.integer("singlemulti") == TREATMENT_CUSTOM```